src/main/java/core/Server.java

```diff
@@ -11,7 +23,7 @@
     private NewClientEventManager tcpListener;
     private UdpEventManager udpEventManager;
     private ServerStrategyContext serverStrategyContext;
-    private List<Client> clientList = Collections.synchronizedList(new ArrayList<>());
+    private List<ClientHandler> clientList = Collections.synchronizedList(new ArrayList<>());
```

Comment:
invoke line 65, this will not work when 2 clients will connect at the same time, might be buggy in future